src/iccpd/src/mlacp_fsm.c

```diff
@@ -0,0 +1,1170 @@
+{
+    if (csm == NULL)
+        return;
+
+    MLACP(csm).sync_req_num = -1;
+    MLACP(csm).need_to_sync = 0;
+    MLACP(csm).error_msg = NULL;
+
+    MLACP(csm).current_state = MLACP_STATE_INIT;
+    memset(MLACP(csm).remote_system.system_id, 0, ETHER_ADDR_LEN);
+
+    MLACP_MSG_QUEUE_REINIT(MLACP(csm).mlacp_msg_list);
+    MLACP_MSG_QUEUE_REINIT(MLACP(csm).arp_msg_list);
+    MLACP_MSG_QUEUE_REINIT(MLACP(csm).mac_msg_list);
+    PIF_QUEUE_REINIT(MLACP(csm).pif_list);
+    LIF_PURGE_QUEUE_REINIT(MLACP(csm).lif_purge_list);
+
+    if (all != 0)
+    {
+        /* if no clean all, keep the arp info & local interface info for next connection*/
+        MLACP_MSG_QUEUE_REINIT(MLACP(csm).arp_list);
+        MLACP_MSG_QUEUE_REINIT(MLACP(csm).mac_list);
+        LIF_QUEUE_REINIT(MLACP(csm).lif_list);
+
+        MLACP(csm).node_id = MLACP_SYSCONF_NODEID_MSB_MASK;
+        MLACP(csm).node_id |= (((inet_addr(csm->sender_ip) >> 24) << 4) & MLACP_SYSCONF_NODEID_NODEID_MASK);
+        MLACP(csm).node_id |= rand() % MLACP_SYSCONF_NODEID_FREE_MASK;
+    }
+
+    return;
+}
+
+/*****************************************
+* MLACP finalize
+*
+* ***************************************/
+void mlacp_finalize(struct CSM* csm)
+{
+    if (csm == NULL)
+        return;
+
+    /* msg destroy*/
+    MLACP_MSG_QUEUE_REINIT(MLACP(csm).mlacp_msg_list);
+    MLACP_MSG_QUEUE_REINIT(MLACP(csm).arp_msg_list);
+    MLACP_MSG_QUEUE_REINIT(MLACP(csm).mac_msg_list);
+    MLACP_MSG_QUEUE_REINIT(MLACP(csm).arp_list);
+    MLACP_MSG_QUEUE_REINIT(MLACP(csm).mac_list);
+
+    /* remove lif & lif-purge queue */
+    LIF_QUEUE_REINIT(MLACP(csm).lif_list);
+    LIF_PURGE_QUEUE_REINIT(MLACP(csm).lif_purge_list);
+    /* remove & destroy pif queue */
+    PIF_QUEUE_REINIT(MLACP(csm).pif_list);
+
+    return;
+}
+
+/*****************************************
+* MLACP FSM Transit
+*
+* ***************************************/
+void mlacp_fsm_transit(struct CSM* csm)
+{
+    struct System* sys = NULL;
+    struct Msg* msg = NULL;
+    static MLACP_APP_STATE_E prev_state = MLACP_SYNC_SYSCONF;
+    ICCHdr* icc_hdr = NULL;
+    ICCParameter* icc_param = NULL;
+    int have_msg = 1;
+
+    if (csm == NULL)
+        return;
+    if ((sys = system_get_instance()) == NULL)
+        return;
+
+    /* torn down event */
+    if (csm->sock_fd <= 0 || csm->app_csm.current_state != APP_OPERATIONAL)
+    {
+        /* drop all legacy mlacp msg*/
+        if (MLACP(csm).current_state != MLACP_STATE_INIT)
+        {
+            MLACP_MSG_QUEUE_REINIT(MLACP(csm).mlacp_msg_list);
+            MLACP_MSG_QUEUE_REINIT(MLACP(csm).arp_msg_list);
+            MLACP_MSG_QUEUE_REINIT(MLACP(csm).mac_msg_list);
+            MLACP(csm).current_state = MLACP_STATE_INIT;
+        }
+        return;
+    }
+
+    /*Update port isolate after 3 seconds*/
+    if (csm->isolate_update_time != 0 && (time(NULL) - csm->isolate_update_time) >= 3)
+    {
+        /*update_peerlink_isolate_from_all_csm_lif(csm);*/
```

Comment:
Based on this logic, we no longer apply a 3-sec delay for port-isolation setting. Should we remove all settings for isolatate_update_time to avoid making unneeded time() system calls?

Reply:
> Based on this logic, we no longer apply a 3-sec delay for port-isolation setting. Should we remove all settings for isolatate_update_time to avoid making unneeded time() system calls?

@lisahnguyen, OK, we will remove these. Thanks.